planner/core/task.go

```diff
@@ -127,7 +127,11 @@ func (t *copTask) finishIndexPlan() {
 	t.indexPlanFinished = true
 	sessVars := t.indexPlan.SCtx().GetSessionVars()
 	// Network cost of transferring rows of index scan to TiDB.
-	t.cst += cnt * sessVars.NetworkFactor * t.tblColHists.GetAvgRowSize(t.indexPlan.Schema().Columns, true)
+	if sessVars.EnableChunkRPC {
```

Comment:
This is not enough

We need to choose a path from
```
IndexScan_TypeDefault
TableScan_TypeDefault
IndexLookup_TypeDefault
IndexScan_TypeChunk
TableScan_TypeChunk
IndexLookup_TypeChunk
```

Reply:
I don't think so. If the plan is decided, then the EncodeType is stable.  
For example, If we want choose a encodeType for a TableScan, calculate cost(TableScan_TypeDefault) and cost(TableScan_TypeChunk), select the smaller one and set the option.